### PR TITLE
Fix #122: Use json template to parse hg log

### LINKED ISSUE #122: Use json template for parsing hg log
Mercurial has a builtin json template: https://book.mercurial-scm.org/read/template.html#parsing-mercurial-output
It is less error prone than specifying the template line manually, and more robust to Mercurial version changes.

### PR DESCRIPTION
**Pull request by hdpoliveira**

### CODE REVIEW

File: src/hg.ts [hunk 1]
```diff
@@ -572,6 +572,19 @@ export interface Commit extends Revision {
 	bookmarks: string[];
 }
 
+export interface NativeCommit {
```
Comment (incidentist) [inline src/hg.ts:575]: Seems like we don't want to export this, since it's intended to only be used internally.

Comment (hdpoliveira) [reply src/hg.ts:575]: oops, copy paste!

**Comment by mrcrowl** [comment]
Sorry! I had a review pending and forgot to hit submit.

**Comment by hdpoliveira** [comment]
The bookmarks is definitely wrong! Thanks for catching that!

I'll remove the comment and check what happens with multi line commits.

**Pull request merged**